pkg/apis/storage/validation/validation.go

```diff
@@ -572,6 +573,21 @@ func validateSELinuxMount(seLinuxMount *bool, fldPath *field.Path) field.ErrorLi
 	return allErrs
 }
 
+// validvalidateServiceAccountTokenInSecrets validates serviceAccountTokenInSecrets and its relation to tokenRequests.
+func validateServiceAccountTokenInSecrets(serviceAccountTokenInSecrets *bool, tokenRequests []storage.TokenRequest, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if serviceAccountTokenInSecrets != nil {
+		if !utilfeature.DefaultFeatureGate.Enabled(features.CSIServiceAccountTokenSecrets) {
```

Comment:
don't check feature gates in validation, clear this field in csiDriverStrategy#PrepareForCreate if the feature is disabled, and in csiDriverStrategy#PrepareForUpdate if the feature is disabled AND the existing object doesn't have the field set already